### DISCUSSION TITLE
Logic for Many-To-One & One-To-Many relations when no items exist

### DISCUSSION THREAD

**Discussion by RageBrahimi** [directus/directus#4757] [Archive]
Hi there!

I'm just trying to work out if my logic is wrong, or there's a potential bug.

I've created two collections: `countries` and `competitions`

In `competitions` there's a `Many to One` field named `country` that is associated with the `countries` collection. I also selected the option to populate the corresponding `One to Many` field named `competitions` in `countries`.

However, when I try make a new item in `countries` I get the following:
```json
{
  "message": "select `competitions`.`countries`, `competitions`.`id` from `competitions` where `competitions`.`countries` in (1) order by `competitions`.`null` asc - ER_BAD_FIELD_ERROR: Unknown column 'competitions.null' in 'order clause'",
  "extensions": {
    "code": "INTERNAL_SERVER_ERROR"
  }
}
```

Obviously at this point I have no item in either `country` or `competition` but either way I cannot make an entry in either as they both rely on a relation. My assumption was `competitions` can be empty when making a new entry in `country`

**Comment by rijkvanzanten** [comment]
Weird! It seems like it's setting the sort value to the `sort_field`, even though it's `null` 🤔 

> order by `competitions`.`null`

@RageBrahimi Can you open a new Issue? (I can't transfer this). Please detail what database driver you're using as well. A db dump with a minimal reproduction would be very helpful 🙂

**Reply by RageBrahimi** [reply]
Sure @rijkvanzanten, will do! 👍

**Reply by RageBrahimi** [reply]
Apologies for the delay! [Here it is](https://github.com/directus/directus/issues/4779).

> Weird! It seems like it's setting the sort value to the `sort_field`, even though it's `null` 🤔
> 
> > order by `competitions`.`null`
> 
> @RageBrahimi Can you open a new Issue? (I can't transfer this). Please detail what database driver you're using as well. A db dump with a minimal reproduction would be very helpful 🙂

**Reply by rijkvanzanten** [reply]
Perfect, thanks 🙂
